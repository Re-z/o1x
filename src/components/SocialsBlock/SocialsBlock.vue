<template>
    <div class="justify-center bg-amber-700 px-2 py-4 text-center">
        <img
            src="https://place-hold.it/100x100"
            class="mx-auto"
        />
        <p class="text-l py-7 text-white">
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quod cumque rem aperiam
            dolores perspiciatis laudantium deleniti id? Quam, enim rerum. Doloribus officiis
            maiores accusantium vero a voluptates tempore quae. Ipsam qui incidunt sunt? Dolores
            labore qui maxime explicabo quaerat deleniti debitis ea adipisci, perferendis veniam
            soluta, nemo nihil illum sit.
        </p>
        <div class="flex justify-center">
            <Button
                data-test-id="socials-button"
                class="px-2"
                :type="BUTTON_TYPES.SECONDARY"
                v-for="social in socials"
                :key="social.title"
                :text="social.title"
                @click="handleSocialClick(social.link)"
            />
        </div>
    </div>
</template>
<script setup lang="ts">
    // imports
    import { ref } from 'vue'
    import Button from '../generic/Button/Button.vue'
    import { BUTTON_TYPES } from '../generic/Button/Button'

    // social buttons click handler
    const socials = ref([
        {
            title: 'Instagram',
            link: 'https://instagram.com',
            img: '',
        },
        {
            title: 'Facebook',
            link: 'https://facebook.com',
            img: '',
        },
        {
            title: 'Threads',
            link: 'https://threads.com',
            img: '',
        },
    ])

    function handleSocialClick(link: any) {
        window.open(link, '_blank')
    }
</script>
