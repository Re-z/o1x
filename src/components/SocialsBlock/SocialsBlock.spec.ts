import { describe, test, expect, vi } from 'vitest'
import { mountComponent } from '@/__tests__/setup'
import SocialsBlock from '@/components/SocialsBlock/SocialsBlock.vue'

describe('SocialsBlock.vue tests', () => {
    test('basic tests', async () => {
        const wrapper = mountComponent(SocialsBlock)
        const socialButton = wrapper.find('[data-test-id="socials-button"]')
        // snapshot
        expect(wrapper.html()).toMatchSnapshot()
        // check if navigation works correctly
        vi.spyOn(window, 'open').mockImplementation(vi.fn())

        await socialButton.trigger('click')
        expect(window.open).toHaveBeenCalled()
    })
})
