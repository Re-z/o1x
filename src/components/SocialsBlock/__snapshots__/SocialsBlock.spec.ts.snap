// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SocialsBlock.vue tests > basic tests 1`] = `
"<div class="justify-center bg-amber-700 px-2 py-4 text-center"><img src="https://place-hold.it/100x100" class="mx-auto">
  <p class="text-l py-7 text-white"> Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quod cumque rem aperiam dolores perspiciatis laudantium deleniti id? Quam, enim rerum. Doloribus officiis maiores accusantium vero a voluptates tempore quae. Ipsam qui incidunt sunt? Dolores labore qui maxime explicabo quaerat deleniti debitis ea adipisci, perferendis veniam soluta, nemo nihil illum sit. </p>
  <div class="flex justify-center">
    <!-- TODO: optimize -->
    <div class="button-wrapper px-2" data-test-id="socials-button">
      <!--v-if-->
      <div class="transform cursor-pointer text-lg transition duration-500 ease-in-out hover:underline active:scale-95" style="color: rgb(255, 255, 255);">Instagram</div>
    </div><!-- TODO: optimize -->
    <div class="button-wrapper px-2" data-test-id="socials-button">
      <!--v-if-->
      <div class="transform cursor-pointer text-lg transition duration-500 ease-in-out hover:underline active:scale-95" style="color: rgb(255, 255, 255);">Facebook</div>
    </div><!-- TODO: optimize -->
    <div class="button-wrapper px-2" data-test-id="socials-button">
      <!--v-if-->
      <div class="transform cursor-pointer text-lg transition duration-500 ease-in-out hover:underline active:scale-95" style="color: rgb(255, 255, 255);">Threads</div>
    </div>
  </div>
</div>"
`;
